### ISSUE TITLE
Stormworks Egg Request

### ISSUE THREAD

**Issue by danymo1221** [parkervcp/eggs#522]
Please fill out the information bellow and remove from the line up
Please understand how Pterodactyl works when you are requesting an egg. (ie. docker-compose doesn't work for a pterodactyl server)
---------------

Service: (Ex. minecraft/factorio/etc)
Stormworks (Windows requires wine)

Does this expand an already existing service: N

Link to game: (Ex. minecraft.net/factorio.com/etc)
Main Website: http://mcro.org/stormworks
Steam: https://store.steampowered.com/app/573090/Stormworks_Build_and_Rescue/

Links for server downloads: This needs to be an official link and not one that is hosted on some forum page or a personal github page.
It can be installed via steamcmd
```
login anonymous
force_install_dir c:\swds\
app_update 1247090 validate

or

steamcmd +login anonymous +force_install_dir c:\swds\ +app_update 1247090 +quit
```

Links for install steps/docs:
https://stormworks.gamepedia.com/Wiki/Guides/Dedicated_Server
This is the official gamepedia docs for installing stormworks
https://steamcommunity.com/sharedfiles/filedetails/?id=1995371855
Or a guide made by someone that's pretty detailed

**Comment by iamkubi** [comment]
I have a Stormworks egg running but need to sort out the flags and figure out how to disable the crazy amount of console spam.  Should be ready for testing in the next couple days.  I don't own the game, so once this issue gets opened with the PR I'd appreciate if you could verify it works.

**Comment by danymo1221** [comment]
Sure I can do that

**Comment by iamkubi** [comment]
Please vote on this issue:
http://mcro.org/issues/view_issue/21739

I may go ahead and add this egg while the console spam issue still exists.  Hopefully they can provide a way to disable or tune it, but it doesn't prevent the server from working.

**Comment by vingard** [comment]
Any updates?

**Comment by iamkubi** [comment]
Looks like the developer closed my feature request as "Rejected" without any update.  I was kind of hoping they'd care since it says to contact them if you have any issues running under Wine, but that doesn't seem to be the case.

The other issue is that I can't get their config file to be editable by the panel.  This is because they cram everything into a single line which makes it incompatible with the existing file parsers.  I had hoped to use either XML or File, but they both have issues.

First an example of the first few lines of a StormWorks config:
```
<?xml version="1.0" encoding="UTF-8"?>
<server_data port="25564" name="Stormworks-Server" seed="6202" save_name="" max_players="32" password="" advanced_mode="true" day_night_length="60" sunrise="0.25" sunset="0.791" infinite_stock="false" unlock_all_islands="false" creative_menu="false" base_island="data/tiles/island12.xml" first_person="false" first_person_vehicle="false" vehicle_damage="false" player_damage="false" npc_damage="false" sharks="false" megalodon="false" teleport_vehicle="true" fast_travel="true" limited_fuel="false" starting_currency="15000" despawn_on_leave="false" unlock_components="true">
        <admins>
                <id value="12345"/>
        </admins>
```

1) XML - The config file is written in XML, so that would be a great parser to use, however the Pterodactyl XML parser only supports replacing the value of elements.  Stormworks uses attributes instead of elements for their config options, which means I can't use the XML parser.

2) File - The file parser lets you change arbitrary stuff, at least I thought it was arbitrary!  Turns out this isn't the case though, it only will perform replacement if the line **starts with** the pattern match.  Unfortunately this means I can't use the file parser either. 
 https://github.com/pterodactyl/daemon/blob/b551500dfa2ca703997f09492fdeab189a9125da/src/helpers/fileparser.js#L75

So as far as I can tell there's no way to automate the settings file updates.  This is a combination of issues with the Stormworks config format and assumptions made by Pterodactyl's config file parser about the way a config file should be formatted.

I had previously been debating various ways to work around this, but I've exhausted the amount of effort I care to expend on this.  I'll see about just pushing the egg requiring any config changes to be done through manual edits, but I'm not thrilled about it.
